src/lib/badge/badge.ts

```diff
@@ -0,0 +1,168 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {Directive, Input, ElementRef, Inject, Optional, NgZone} from '@angular/core';
+import {coerceBooleanProperty} from '@angular/cdk/coercion';
+import {ThemePalette} from '@angular/material/core';
+import {AriaDescriber} from '@angular/cdk/a11y';
+import {DOCUMENT} from '@angular/common';
+
+let nextId = 0;
+
+export type MatBadgePosition = 'above after' | 'below after' | 'above before' | 'above after';
+export type MatBadgeSize = 'small' | 'medium' | 'large';
+
+/** Directive to display a text badge. */
+@Directive({
+  selector: '[matBadge]',
+  host: {
+    'class': 'mat-badge',
+    '[class.mat-badge-overlap]': '_overlap',
+    '[class.mat-badge-above]': 'isAbove()',
+    '[class.mat-badge-below]': '!isAbove()',
+    '[class.mat-badge-before]': '!isAfter()',
+    '[class.mat-badge-after]': 'isAfter()',
+    '[class.mat-badge-small]': 'size === "small"',
+    '[class.mat-badge-medium]': 'size === "medium"',
+    '[class.mat-badge-large]': 'size === "large"',
+    '[class.mat-badge-hidden]': 'hidden',
+  },
+})
+export class MatBadge {
+
+  /** The color of the badge. Can be `primary`, `accent`, or `warn`. */
+  @Input('matBadgeColor')
```

Comment:
Question: The reason that you can't just use the `mixinColor` here is because the input has an alias?

Reply:
It needs to have a separate alias for color since the host can also describe a color.